File: Algorithms/Warmup/comparethetriplets/comparethetriplets.py
#!/bin/python3

import math
import os
import random
import re
import sys

def compareTriplets(a, b):
    """
    The function takes in two lists a and b. It compares the integers in the lists and returns a list containing the number of times the integers in list a are greater than the integers in list b and the number of times the integers in list b are greater than the integers in list a.
    :param a: a list of integers
    :param b: a list of integers
    :return: a list of 2 integers
    """
    alice = 0
    bob = 0
    for i in range(3):
        if a[i] > b[i]:
            alice += 1
        elif a[i] < b[i]:
            bob += 1
    return [alice, bob]

    

if __name__ == '__main__':
    fptr = open(os.environ['OUTPUT_PATH'], 'w')

    a = list(map(int, input().rstrip().split()))

    b = list(map(int, input().rstrip().split()))

    result = compareTriplets(a, b)

    fptr.write(' '.join(map(str, result)))
    fptr.write('\n')

    fptr.close()

File: Algorithms/Warmup/comparethetriplets/comparethetriplets.r


#
# Complete the 'compareTriplets' function below.
#
# The function is expected to return an INTEGER_ARRAY.
# The function accepts following parameters:
#  1. INTEGER_ARRAY a
#  2. INTEGER_ARRAY b
#

compareTriplets <- function(a, b) {
    # This function compares two arrays and returns the number of elements that are greater in the first array than the second array and vice versa.
    # :param a: An array of integers
    # :param b: An array of integers
    # :return: An array of two integers

    alice <- 0
    bob <- 0
    
    for (i in 1:3) {
        if (a[i] > b[i]) {
            alice <- alice + 1
        } else if (a[i] < b[i]) {
            bob <- bob + 1
        }
    }
    
    return(c(alice, bob))
}

stdin <- file('stdin')
open(stdin)

fptr <- file(Sys.getenv("OUTPUT_PATH"))
open(fptr, open = "w")

a <- strsplit(trimws(readLines(stdin, n = 1, warn = FALSE), which = "right"), " ")[[1]]
a <- as.integer(a)

b <- strsplit(trimws(readLines(stdin, n = 1, warn = FALSE), which = "right"), " ")[[1]]
b <- as.integer(b)

result <- compareTriplets(a, b)

writeLines(paste(result, collapse = " "), con = fptr)

close(stdin)
close(fptr)
